app/views/protocol_importers/_preview_modal_footer.html.erb

```diff
@@ -14,7 +14,7 @@
         <%= label_tag :default_public_user_role_id, t("protocols.new_protocol_modal.role_label") %>
         <% default_role = UserRole.find_by(name: I18n.t('user_roles.predefined.viewer')).id %>
         <%= hidden_field_tag :default_public_user_role_id, value: default_role %>
-        <%= select_tag :role_selector, options_for_select(user_roles_collection(@protocol), default_role) %>
+        <%= select_tag :role_selector, options_for_select(user_roles_collection.select {|ur| ur[0] != "Technician" }, default_role) %>
```

Comment:
Do not filter here. user_roles_collection filter for you which roles will be shown in the dropdown based on permissions.